### ISSUE TITLE
test_cim_obj.py fails with XMLSyntaxError: Char 0x0 out of allowed range

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#2337]
This showed up recently on Travis using macOS with the Travis xcode12 image, upgraded to Python 3.8 and with minimum package levels.

Example Travis run: https://travis-ci.org/github/pywbem/pywbem/jobs/707527030

Experiments show that the following reproduces this issue:
```
$ python -c "from lxml import etree; xs2='<PROPERTY NAME=\"Foo\U00010142\" TYPE=\"string\"/>'; etree.fromstring(xs2)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "src/lxml/etree.pyx", line 3236, in lxml.etree.fromstring
  File "src/lxml/parser.pxi", line 1876, in lxml.etree._parseMemoryDocument
  File "src/lxml/parser.pxi", line 1757, in lxml.etree._parseDoc
  File "src/lxml/parser.pxi", line 1068, in lxml.etree._BaseParser._parseUnicodeDoc
  File "src/lxml/parser.pxi", line 601, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 711, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 640, in lxml.etree._raiseParseError
  File "<string>", line 1
lxml.etree.XMLSyntaxError: Char 0x0 out of allowed range, line 1, column 2
```

This error shows up with lxml versions 4.4.1 (our current minimum version) and 4.4.2, but it succeeds with 4.4.3.

**Comment by andy-maier** [comment]
Looking at the changes in lxml 4.4.3 since 4.4.2 ([change log](https://lxml.de/4.4/changes-4.4.3.html), which is consistent with the [git repo changes](https://github.com/lxml/lxml/commits/lxml-4.4)), it does not become clear how this could be the reason for this particular error coming in and out. But the behavior is repeatable when changing just the version of the lxml package back and forth.

**Comment by andy-maier** [comment]
Addressed i PR #2338
